### PR TITLE
Remove some unused obsolete stuff

### PR DESCRIPTION
**Pull request by Visne**
Requires space-wizards/space-station-14#4657.

### CODE REVIEW

File: Robust.Client/UserInterface/Controls/GridContainer.cs [hunk 1]
```diff
@@ -84,55 +84,12 @@ public int Rows
...
-        {
-            set => MaxGridWidth = value;
-        }
 
         public float MaxGridWidth
         {
             set => SetMaxSize(Dimension.Column, value);
         }
 
-        /// <summary>
```
Comment (mirrorcult) [inline Robust.Client/UserInterface/Controls/GridContainer.cs]: these xmldocs seem valuable, put them on the non-obsolete property instead? (same goes for maxwidth)

File: Robust.Shared/GameObjects/Components/ITransformComponent.cs [hunk 2]
```diff
@@ -89,11 +89,6 @@ public interface ITransformComponent : IComponent
         /// </summary>
         public EntityUid ParentUid { get; set; }
 
-        /// <summary>
-        /// Whether or not this entity is on the map, AKA it has no parent.
-        /// </summary>
-        bool IsMapTransform { get; }
```
Comment (mirrorcult) [inline Robust.Shared/GameObjects/Components/ITransformComponent.cs:95]: why is this removed?

File: Robust.Client/UserInterface/Controls/GridContainer.cs (hunk 1 above)
Comment (Visne) [reply Robust.Client/UserInterface/Controls/GridContainer.cs]: Done.
